### ISSUE TITLE
Attach to running iOS application already running Xcode debugger

### ISSUE THREAD

**Issue by ben-xD** [flutter/flutter-intellij#5724]
Is there a way to attach Android Studio/ IntelliJ to the running iOS application?

I have tried running the iOS application through Android Studio, and then trying to `Xcode > Debug > Attach to process > Runner`, and I got the error:
![Screenshot 2021-08-25 at 07 36 39](https://user-images.githubusercontent.com/24711048/130738868-f1eb2141-4503-4d30-a8a9-1b08f9ab5755.png)

```
error: attach by pid '425' failed -- tried to attach to process already being debugged
```

This would be great as it would allow debugging the iOS and dart code simultaneously, like we can do for Android Applications (by attaching to the Android process in Android Studio).

**Comment by ben-xD** [comment]
Thanks for your responses 🙂, I'd briefly seen `flutter attach` before but never used it. Nice to finally understand/ use it. And then I found the `flutter attach` button right next to the `attach debugger to Android process`, its been there all this time 🙈

What worked for me a few times was:
- Run in Xcode (if anyone wants the dart define variables to pass through, build it in Android Studio/ command line first, then rebuild in Xcode)
- Attach using Android Studio's `Run > Flutter Attach`

---

However, I faced some issues:
- After the initial successful debug, rebuilding on Xcode and using Android Studio or running `flutter attach` again, it was just stuck on `Waiting for a connection from Flutter on iPhone SE BB🔫...`. I had to reinstall the app to fix this, but reinstalling doesn't actually always work/ fix the problem. I tried this 3 times, and only once this uninstalling/ reinstalling the app helped.

---
A separate issue:

- When you attach and detach, another application becomes listed as an option to `flutter attach`. When I repeated this (e.g. attached, reattached, attached, reattached), and I had 3 applications that I could attach to:
```
There are multiple observatory ports available.
Rerun this command with one of the following passed in as the appId:

  flutter attach --app-id my.bundle.id
  flutter attach --app-id my.bundle.id (2)
  flutter attach --app-id my.bundle.id (3)
```

This means @jmagman 's suggestions regarding deattaching (pressing `d`) won't work on Android Studio, since Android Studio doesn't allow you to pass `-app-id` to flutter attach.

**Comment by stevemessick** [comment]
@ben-xD, there may be a problem with stopping your app. When using multiple apps to control your Flutter app you'll need to be careful to ensure the control processes are not running when you re-launch your app. It looks like the Mac had multiple `flutter` tools running. Make sure those are stopped before launching your app.

**Comment by ben-xD** [comment]
I don't remember having any flutter processes running (that was visible at least). Apart from Android Studio (the debugger was completely stopped/ closed) when I ran Xcode.

I tried it again, launching the app through Xcode, followed by (in Android Studio) attaching (success), detaching and attaching causes:
```
There are multiple observatory ports available.
Rerun this command with one of the following passed in as the appId:

  flutter attach --app-id my.bundle.id
  flutter attach --app-id my.bundle.id (2)
```

I don't have any terminals open. 

However, when rebuild the app in Xcode, Android Studio can successfully attach to it again. Once something attaches to the application, it seems like something doesn't get cleaned up correctly on detachment?  

At this point, it doesn't affect me since my dart-define variables are saved once I build at least once with Android Studio/ `flutter` cli, and therefore I don't need to use the `flutter` cli. But it would affect someone who needs to use the flutter CLI, since once you detach, you **can't reattach in Android Studio**. You can still reattach/ detach as many times using the `flutter` cli though.

**Comment by stevemessick** [comment]
It seems to me the problem is in the `flutter` tool. The detach/attach sequence appears to be starting a new app instead of re-using the old one, or just stopping it. You might want to discuss that in the Flutter SDK [issues](https://github.com/flutter/flutter/issues).
